Lib/types.py

```diff
@@ -143,6 +143,35 @@ def _calculate_meta(meta, bases):
                         "of the metaclasses of all its bases")
     return winner
 
+
+def get_original_bases(cls, /):
+    """
+    Return the objects in the bases list in the class's definition before
+    they could have been modified by ``__mro_entries__``. This is useful for
+    introspecting Generics.
+
+    Examples::
+
+        from typing import TypeVar, Generic
+
+        T = TypeVar("T")
+
+        class Foo(Generic[T]): ...
+
+        class Bar(Foo[int], float): ...
```

Comment:
Don't think we need these newlines:

```suggestion
        T = TypeVar("T")
        class Foo(Generic[T]): ...
        class Bar(Foo[int], float): ...
```